src/es-module-shims.js

```diff
@@ -52,17 +52,20 @@ async function importMetaResolve (id, parentUrl = this.url) {
   return resolve(id, `${parentUrl}`);
 }
 
-Object.defineProperties(importShim, {
+let esmsState = {}
+self.esmsState = esmsState
```

Comment:
Of course this has to be exposed, sorry I overlooked that here.

Sorry to be difficult but I think we should probably just leave it as `importShim` then if we need a global anyway since that's to hand. Alternatively a private name like `_esmsState`.

Reply:
Understood, `_esmsState` sounds like the better option to me since it at least allows us to separate the public API from the internal state, even if it's still technically accessible on the global scope.

Addressed here: https://github.com/guybedford/es-module-shims/pull/109/commits/348e65b2f902afeb0e2361c421276e72247a45d5